build/common.sh

```diff
@@ -472,7 +472,11 @@ function kube::build::build_image() {
   kube::version::save_version_vars "${build_context_dir}/kube-version-defs"
 
   cp build/build-image/Dockerfile ${build_context_dir}/Dockerfile
-  sed -i "s/KUBE_BUILD_IMAGE_CROSS/${KUBE_BUILD_IMAGE_CROSS}/" ${build_context_dir}/Dockerfile
+  if kube::build::is_osx; then
+    sed -i "" "s/KUBE_BUILD_IMAGE_CROSS/${KUBE_BUILD_IMAGE_CROSS}/" ${build_context_dir}/Dockerfile
```

Comment:
sed -i worked on both GNU and BSD however "="(and any other symbol between -i and "" ) will be treated as part for the backup suffix, backup file "Dockerfile=" will be generated on both OSes.
Here is examples
1. If no backup required, we can run the command like this, the difference is on BSD, a space is mandatory  between -i and "".
On GNU
sed -i"" "s/aa/bb/" data
On BSD
sed -i "" "s/aa/bb/" data
2. If we run the command like what you suggested, backup files "data=" on both OSes are generated
sed -i="" "s/aa/bb/" data

You can confirm this on your GNU, thanks. 
Before I push the changes, I did a lots of search and I did not see a compatible way for both BSD and GNU.